cliquet/storage/postgresql/__init__.py

```diff
@@ -103,31 +103,46 @@ class PostgreSQL(PostgreSQLClient, StorageBase):
 
     """
 
+    schema_version = 3
+
     def __init__(self, *args, **kwargs):
         self._max_fetch_size = kwargs.pop('max_fetch_size')
         super(PostgreSQL, self).__init__(*args, **kwargs)
         self._init_schema()
 
+    def _execute_sql_file(self, filepath):
+        here = os.path.abspath(os.path.dirname(__file__))
+        schema = open(os.path.join(here, 'schema.sql')).read()
+        with self.connect() as cursor:
+            cursor.execute(schema)
+
     def _init_schema(self):
-        """Create PostgreSQL tables, only if not exists.
+        """Create PostgreSQL tables, and run necessary schema migrations.
 
         :note:
             Relies on JSON fields, available in recent versions of PostgreSQL.
         """
-        # Since indices cannot be created with IF NOT EXISTS, inspect.
-        query = """
-        SELECT *
-          FROM pg_tables
-         WHERE tablename = 'records';
-        """
-        with self.connect() as cursor:
-            cursor.execute(query)
-            exists = cursor.rowcount > 0
-
-        if exists:
-            logger.debug('Detected PostgreSQL storage tables')
+        version = self._get_installed_version()
+        if not version:
+            # Create full schema.
+            self._check_database_encoding()
+            self._execute_sql_file('schema.sql')
+            logger.info('Created PostgreSQL storage tables.')
             return
 
+        logger.debug('Detected PostgreSQL schema version %s.' % version)
+        migrations = [(v, v + 1) for v in range(version, self.schema_version)]
+        if not migrations:
+            logger.info('Schema is up-to-date.')
+
+        for migration in migrations:
+            logger.info('Migrate schema from version %s to %s.' % migration)
+            filepath = 'migration_%s_%s.sql' % migration
+            self._execute_sql_file(filepath)
```

Comment:
As @magopian pointed out, adding an assertion on the current version would be useful, to prevent skipping numbers